docs/Rest-API.md

```diff
@@ -0,0 +1,147 @@
+# Axe REST API
+
+Documentation of the REST API that comes along with the Axe service.
+
+## AxeResult
+
+This endpoint serves as a wrapper to collect all of the information from an axe run. 
+
+### Motivation
+
+- Provide access to results and context easily.
+- Avoid timing issues that come with multiple requests.
+
+### Simplified Sample
+```
+{
+  "axeContext" : { @AxeContext },
+  "axeResult" : { AXRL },
+  "axeConf" : { TBD by Framework }  
+}
+```
+
+### Requirements
+
+- Data points are collected at the same point in time, or as close as can be expected.
+
+## AxeContext
+
+Wraps data together that is needed to identify a particular View, with a particular screenshot,
+for a particular device configuration.
+
+### Motivation
+
+- Collect View information in one place.
+- Allow for direct correlation of screenshot and ViewHierarchy
+- Easy developer debugging.
+- Reproducability of results.
+
+### Simplified Sample
+
+```
+{
```

Comment:
Should we include any meta data about the scan itself, e.g., version of Axe, duration, errors? This information can be useful for posterity.

Reply:
Hmm. My question is where does this belong. At this level, or one level up. Good idea though. We will add this.